### PR TITLE
Refactoring and cleanup

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Issues? | https://github.com/yiisoft/db/issues/154 https://github.com/yiisoft/db/issues/108 https://github.com/yiisoft/db/issues/152

- [x] Add testsuite db-mssql.
- [x] Add testsuite db-pgsql.
- [x] Add testsuite db-mysql.
- [x] Add testsuite db-sqlite.

### CODE REVIEW

File: src/Helper/Dsn.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function getDsn(): string
         $this->dsn = "$this->driver:" . "host=$this->host" . ';' . "dbname=$this->dbname";
```
Comment (samdark) [inline src/Helper/Dsn.php:42]: The Dsn itself isn't a helper but a value object and belongs to `Yiisoft\Db\Connection` namespace.

File: src/Helper/Dsn.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ public function getDsn(): string
...
 
         if ($this->port !== null) {
-            $this->dsn = $this->dsn . ';' . "port=$this->port";
+            $this->dsn .= ';' . "port=$this->port";
```
Comment (samdark) [inline src/Helper/Dsn.php:45]: `$this->dsn` can be replaced with local variable.

File: src/Connection/Connection.php [hunk 3]
```diff
@@ -160,7 +166,7 @@
  * @property Transaction|null $transaction The currently active transaction. Null if no active transaction. This
```
Comment (samdark) [inline src/Connection/Connection.php:166]: `@property` annotations seem become obsolete and could be deleted.

File: src/Connection/ConnectionInterface.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Connection;
+
+use Yiisoft\Db\Command\Command;
+use Yiisoft\Db\Exception\Exception;
+use Yiisoft\Db\Exception\InvalidConfigException;
+use Yiisoft\Db\Schema\Schema;
+
+interface ConnectionInterface
```
Comment (samdark) [inline src/Connection/ConnectionInterface.php:12]: Should it contain more methods such as the following?

- beginTransaction(), getTransaction()
- open(), close(), isActive()

File: src/Helper/Dsn.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Helper/Dsn.php:42]: Fixed.

File: src/Helper/Dsn.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Helper/Dsn.php]: Fixed.

**Pull request merged**